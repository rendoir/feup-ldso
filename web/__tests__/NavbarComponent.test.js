import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../src/NavbarComponent';
import { mount } from 'enzyme';
import sinon from 'sinon';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

var mockAxios = new MockAdapter(axios);

describe("Logout", () => {

    it("Check logout is successful", async() => {
        document.cookie = "access_token=12345";

        let spy = sinon.spy(Navbar.prototype, 'getTokenFromCookie');

        const wrapper = mount(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>);

        let wrapperNavbar = wrapper.find(Navbar).first();
        wrapperNavbar.instance().getTokenFromCookie = jest.fn(() => "");

        mockAxios.onPost('http://localhost:3030/logout').reply(200);

        const button = wrapper.find('button#logout-button').first();
        button.simulate('click');

        expect(spy.calledOnce);

    });

    it("Check getTokenFromCookie is successful", async() => {
        document.cookie = "access_token=12345";

        const wrapper = mount(
            <BrowserRouter>
                <Navbar />
            </BrowserRouter>);

        let wrapperNavbar = wrapper.find(Navbar).first();
        let result = wrapperNavbar.instance().getTokenFromCookie();

        expect(result).toEqual("12345");

    });


});
