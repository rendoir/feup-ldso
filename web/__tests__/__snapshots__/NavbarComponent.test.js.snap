// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elements 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
  id="navbar-custom"
>
  <a
    className="navbar-brand"
    href="#"
  >
    <a
      aria-current={null}
      className="navbar-brand"
      href="/events"
      onClick={[Function]}
    >
      <img
        className=""
        id="logo"
        src="/logo.png"
      />
    </a>
  </a>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    />
  </div>
</nav>
`;

exports[`renders elements 2`] = `null`;

exports[`renders elements 3`] = `null`;
